aiida/orm/nodes/data/code/containerized.py

```diff
@@ -0,0 +1,167 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Data plugin representing an executable code on a remote computer.
+
+This plugin should be used if an executable is pre-installed on a computer. The ``InstalledCode`` represents the code by
+storing the absolute filepath of the relevant executable and the computer on which it is installed. The computer is
+represented by an instance of :class:`aiida.orm.computers.Computer`. Each time a :class:`aiida.engine.CalcJob` is run
+using an ``InstalledCode``, it will run its executable on the associated computer.
+"""
+import click
+
+from aiida.common.lang import type_check
+
+from .installed import InstalledCode
+from .portable import PortableCode
+
+__all__ = (
+    'InstalledContainerizedCode',
+    'PortableContainerizedCode',
+)
+
+
+class Containerized:
+    """Containerized Mixin for code that can run on container"""
+    _KEY_ATTRIBUTE_ENGINE_COMMAND: str = 'engine_command'
+    _KEY_ATTRIBUTE_IMAGE: str = 'image'
```

Comment:
Maybe use:
```suggestion
    _KEY_ATTRIBUTE_IMAGE_NAME: str = 'image_name'
```
Not sure if in the future we might need something like `image_version`. In any case, I think this is more clear right?

Reply:
I'd say image=image_name:image_version (`ubuntu:kinetic`)? `image` here is more generic and more flexible for users to attach whatever they like.